s3client.go

```diff
@@ -184,8 +184,18 @@ func (client *s3client) UploadFile(bucketName string, remotePath string, localPa
 	}
 
 	defer localFile.Close()
+	
+	// Automatically adjust partsize for larger files.
+	fSize := stat.Size()
+	if fSize > int64(uploader.MaxUploadParts) * uploader.PartSize {
+		partSize := fSize / int64(uploader.MaxUploadParts)
+		if fSize % int64(uploader.MaxUploadParts) != 0 {
+			partSize++
```

Comment:
What's the idea behind this bit?

Reply:
The first condition checks whether the file is larger than the maximum size the uploader can handle (based on the default part size and maxuploadparts).

If so, it calculates the new part size so that the maximum is increased to accomodate it.

The second condition checks for a remainder of the integer division - which is likely there - and increases the partSize by 1 to ensure the maximum also accomodates this remainder (otherwise the maximum would fall short by 1-MaxUploadParts bytes).

Reply:
Is it better logging out a message when partsize is adjusted? The log message could include filesize, new partsize, etc.